### PR TITLE
predictions vs. outcomes graph and toggles

### PR DESCRIPTION
**Pull request by MariaC27**
# predictions vs. outcomes graph and toggles

add predictions vs outcomes graph on prediction tabs, toggles, bugfixes

## Type of Change

- [x] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Documentation update

## Tickets

- [554](https://app.zenhub.com/workspaces/pine-beetle-prediction-tool-5d8ab1b0022c8d0001c05ba8/issues/dali-lab/pine-beetle-frontend/554)
- [555](https://app.zenhub.com/workspaces/pine-beetle-prediction-tool-5d8ab1b0022c8d0001c05ba8/issues/dali-lab/pine-beetle-frontend/555)

## Screenshots

Capture

### CODE REVIEW

File: src/screens/prediction/component.js [hunk 1]
```diff
@@ -28,11 +36,18 @@ const Prediction = (props) => {
...
+    clearAllSelections,
   } = props;
 
   const [showAnimation, setShowAnimation] = useState(true);
+  const isGraphView = chartMode === CHART_MODES.GRAPH;
+  const setGraphView = () => setChartMode(CHART_MODES.GRAPH);
+  const setMapView = () => setChartMode(CHART_MODES.MAP);
 
   useEffect(() => {
+    clearAllSelections(); // clears selections initially when switching to this tab
```
Comment (jeffzyliu) [inline src/screens/prediction/component.js:52]: Does this also trigger when switching to this tab for the second time? I cant remember how useEffect works in that instance

File: src/state/actions/selections.js [hunk 2]
```diff
@@ -101,12 +102,22 @@ export function getAvailableStates(overrideFilter = {}, { historical = true, pre
...
 
       if (prediction) {
-        const predictionStates = await api.getAvailableStates(dataMode, {
-          ...filters,
-          isPrediction: true,
-          startYear: filters.predictionYear,
-          endYear: filters.predictionYear,
-        });
+        let predictionStates = [];
+        if (getState().selections.chartMode === CHART_MODES.MAP) {
```
Comment (jeffzyliu) [inline src/state/actions/selections.js:106]: Would recommend considering the refactors I talked about earlier, might help keeping things clean

File: src/state/reducers/data.js [hunk 3]
```diff
@@ -16,6 +16,8 @@ const initialState = {
 
 const DataReducer = (state = initialState, action) => {
   switch (action.type) {
+    // case ActionTypes.SET_START_YEAR:
+    // break;
```
Comment (jeffzyliu) [inline src/state/reducers/data.js]: Will this be used later?

**Review by jeffzyliu — APPROVED** [review]
Glad you were able to get things working!

File: src/screens/prediction/component.js (hunk 1 above)
Comment (MariaC27) [reply src/screens/prediction/component.js:52]: I think it does

File: src/state/reducers/data.js (hunk 3 above)
Comment (MariaC27) [reply src/state/reducers/data.js]: don't think so, will remove

File: src/app.js [hunk 4]
```diff
@@ -97,7 +104,7 @@ const App = (props) => {
               <Route exact path={ROUTES.HOME} component={Home} />
               <Route path={ROUTES.ABOUT} component={About} />
               <Route path={ROUTES.ADMIN} component={Admin} />
-              <Route path={ROUTES.TRAPPING_DATA} component={TrappingData} />
+              <Route path={ROUTES.TRAPPING_DATA} component={TrappingData} onEnter={() => console.log('switch')} />
```
Comment (AlejandroALopez) [inline src/app.js]: might want to remove this console log

**Review by AlejandroALopez — APPROVED** [review]

**Pull request merged**